plugins/woocommerce/src/Internal/Admin/Emails/EmailListingRestController.php

```diff
@@ -0,0 +1,195 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Admin\Emails;
+
+use Automattic\WooCommerce\Internal\RestApiControllerBase;
+use Automattic\WooCommerce\Internal\EmailEditor\WCTransactionalEmails\WCTransactionalEmails;
+use Automattic\WooCommerce\Internal\EmailEditor\WCTransactionalEmails\WCTransactionalEmailPostsGenerator;
+use WP_Error;
+use WP_REST_Request;
+
+/**
+ * Controller for the REST endpoint to send an email preview.
+ */
+class EmailListingRestController extends RestApiControllerBase {
+
+	/**
+	 * Email listing nonce.
+	 *
+	 * @var string
+	 */
+	const NONCE_KEY = 'email-listing-nonce';
+
+	/**
+	 * The root namespace for the JSON REST API endpoints.
+	 *
+	 * @var string
+	 */
+	protected string $route_namespace = 'wc-admin-email';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected string $rest_base = 'settings/email/listing';
+
+	/**
+	 * Email template generator instance.
+	 *
+	 * @var WCTransactionalEmailPostsGenerator
+	 */
+	private $email_template_generator;
+
+	/**
+	 * Get the WooCommerce REST API namespace for the class.
+	 *
+	 * @return string
+	 */
+	protected function get_rest_api_namespace(): string {
+		return 'wc-admin-email';
+	}
+
+	/**
+	 * The constructor.
+	 */
+	public function __construct() {
+		$this->email_template_generator = new WCTransactionalEmailPostsGenerator();
+		add_action( 'woocommerce_init', [ $this->email_template_generator, 'init_default_transactional_emails' ] );
+	}
+
+	/**
+	 * Register the REST API endpoints handled by this controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->route_namespace,
+			'/' . $this->rest_base . '/recreate-email-post',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => fn( $request ) => $this->recreate_email_post( $request ),
+					'permission_callback' => fn( $request ) => $this->check_permissions( $request ),
+					'args'                => $this->get_args_for_recreate_email_post(),
+					'schema'              => $this->get_schema_with_message(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get the accepted arguments for the POST recreate-email-post request.
+	 *
+	 * @return array[]
+	 */
+	private function get_args_for_recreate_email_post() {
+		return array(
+			'emailId'  => array(
```

Comment:
I think the parameter `emailId` should be in the snakecase format for alignment with the rest of the codebase. So I would expect here `email_id`. What do you think?

Reply:
Okay 👌🏿

Reply:
Implemented in https://github.com/woocommerce/woocommerce/pull/57827/commits/47c1bdf62a6d1ee99118da850a03158bf30bbf9f